src/base/bittorrent/torrentcreatorthread.cpp

```diff
@@ -160,3 +160,13 @@ void TorrentCreatorThread::run()
         emit creationFailure(QString::fromStdString(e.what()));
     }
 }
+
+int TorrentCreatorThread::calculateTotalPieces(const QString &inputPath, const int pieceSize)
+{
+    if (inputPath.isEmpty())
+        return 0;
+
+    libt::file_storage fs;
+    libt::add_files(fs, Utils::Fs::toNativePath(inputPath).toStdString(), fileFilter);
+    return libt::create_torrent(fs, pieceSize).num_pieces();
```

Comment:
Did you benchmark to see if it is quicker to add ourselves the filesizes and divide by the piece size to get the piece count?
We could even cache the filesizes so there wouldn't be any delay when user chooses a different piece size...